<launch>

	<arg name="rviz" default="false" />
 	<!-- <param name="use_sim_time" value="true" /> -->
	<arg name="image_topic" default="/left_camera/image" />
	<arg name="undistorted_image_topic" default="/left_camera/undistorted_image" />
	<arg name="lidar_topic" default="/ouster/points" />
	<arg name="imu_topic" default="/ouster/imu" />
	<arg name="gnss_topic" default="/gps/fix" />

	<node pkg="fast_lio_sam" type="fast_lio_sam_sensor_recorder" name="fast_lio_sam_sensor_recorder" output="screen">
		<param name="base_directory" value="/home/kodifly/workspaces/fastliosam_ws/src/FAST-LIO-SAM/fast_lio_sam/data" />
		<param name="send_json" value="false" />
		<remap from="/image_topic" to="$(arg image_topic)"/>
		<remap from="/undistorted_image_topic" to="$(arg undistorted_image_topic)"/>
		<remap from="/lidar_topic" to="$(arg lidar_topic)"/>
		<remap from="/imu_topic" to="$(arg imu_topic)"/>
		<remap from="/gnss_topic" to="$(arg gnss_topic)"/>
	</node>
	
	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_sam)/config/data_rviz.rviz" />
	</group>
	
	<node pkg="image_transport" type="republish" name="republish" args="compressed in:=/hik_camera/image raw out:=/left_camera/image" output="screen" respawn="true"/>
</launch> 	
